### PR TITLE
Graduate PodSchedulingReadiness to beta

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind feature
/sig scheduling

#### What this PR does / why we need it:

1. Promote `PodSchedulingReadiness` to beta
2. Refine the API comments requested here: https://github.com/kubernetes/kubernetes/pull/113274#discussion_r1013374289

#### Which issue(s) this PR fixes:

Part of #113608

#### Does this PR introduce a user-facing change?

```release-note
PodSchedulingReadiness is graduated to beta.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/3521-pod-scheduling-readiness
```

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -3386,14 +3386,19 @@ type PodSpec struct {
 	HostUsers *bool `json:"hostUsers,omitempty" protobuf:"bytes,37,opt,name=hostUsers"`
 
 	// SchedulingGates is an opaque list of values that if specified will block scheduling the pod.
-	// More info:  https://git.k8s.io/enhancements/keps/sig-scheduling/3521-pod-scheduling-readiness.
+	// If a pod carry non-empty schedulingGates, it will stay in the SchedulingGated state to prevent scheduler
```
Comment (Huang-Wei) [inline staging/src/k8s.io/api/core/v1/types.go:3389]: Rewording the comments per https://github.com/kubernetes/kubernetes/pull/113274#discussion_r1013374289.

@smarterclayton I'd super appreciate it if have cycles to review the new comments.

**Comment by liggitt** [comment]
if we're changing validation (https://github.com/kubernetes/kubernetes/pull/115821), beta promotion should happen in 1.28, right?

**Comment by Huang-Wei** [comment]
> if we're changing validation (#115821), beta promotion should happen in 1.28, right?

@ahg-g and I didn't expect #115821 to postpone it to 1.28... See https://github.com/kubernetes/kubernetes/pull/115821#discussion_r1108899923

**Comment by liggitt** [comment]
maybe I was thinking of that backwards… that PR is tightening validation, so anything it allows in under beta the previous version in alpha will be equally happy with? If so, that doesn't require any additional delay

**Comment by Huang-Wei** [comment]
> that PR is tightening validation, so anything it allows in under beta the previous version in alpha will be equally happy with?

I _think_ so - given scheduling gates can only be removed. I did a test:

- Create a pod with a non-qualified scheduling gate name in 1.26
- Bump version to 1.27, removing the scheduling gate, it functions well as ValidatePodUpdate doesn't do general (the tightened logic in 1.27) validation:
 
 
 
 https://github.com/kubernetes/kubernetes/blob/f09eeaf104fe65a911dbd01a0e004b18663d9cee/pkg/apis/core/validation/validation.go#L4543

**Comment by liggitt** [comment]
Unit test failure looks related (looks like the scheduler config defaults adjust when this feature is enabled)

This should drop the feature and alpha tag from the e2e test so that it runs in the normal e2e job:

> ginkgo.It("validates Pods with non-empty schedulingGates are blocked on scheduling [Feature:PodSchedulingReadiness] [alpha]", func(ctx context.Context) {

**Comment by liggitt** [comment]
Otherwise, the PR lgtm

File: api/openapi-spec/v3/api__v1_openapi.json [hunk 2]
```diff
@@ -5209,7 +5209,7 @@
             "type": "string"
           },
           "schedulingGates": {
-            "description": "SchedulingGates is an opaque list of values that if specified will block scheduling the pod. More info:  https://git.k8s.io/enhancements/keps/sig-scheduling/3521-pod-scheduling-readiness.\n\nThis is an alpha-level feature enabled by PodSchedulingReadiness feature gate.",
+            "description": "SchedulingGates is an opaque list of values that if specified will block scheduling the pod. If a pod carry non-empty schedulingGates, it will stay in the SchedulingGated state to prevent scheduler perform pointless scheduling attempts.\n\nSchedulingGates can only be specified in pod creation time, and be removed-only afterwards.\n\nThis is a beta feature enabled by PodSchedulingReadiness feature gate.",
```
Comment (ahg-g) [inline api/openapi-spec/v3/api__v1_openapi.json:5212]:
```suggestion
            "description": "SchedulingGates is an opaque list of values that if specified will block scheduling the pod. If schedulingGates is not empty, the pod will stay in the SchedulingGated state and the scheduler will not attempt to schedule the pod.\n\nSchedulingGates can only be set at pod creation time, and be removed only afterwards.\n\nThis is a beta feature enabled by the PodSchedulingReadiness feature gate.",
```

Comment (Huang-Wei) [reply api/openapi-spec/v3/api__v1_openapi.json:5212]: Updated.

**Comment by Huang-Wei** [comment]
> Unit test failure looks related (looks like the scheduler config defaults adjust when this feature is enabled)

Yes, I fixed those UT failures.

> This should drop the feature and alpha tag from the e2e test so that it runs in the normal e2e job:

Oh, I only removed the alpha tag. Let me remove the `Feature` tag as well.

**Comment by Huang-Wei** [comment]
The CI failure doesn't seem related. All the alpha tests have passed:

```
I0218 03:53:34.523] Ran 40 of 7083 Specs in 2202.432 seconds
I0218 03:53:34.523] SUCCESS! -- 40 Passed | 0 Failed | 0 Pending | 7043 Skipped
I0218 03:53:34.524] PASS
```

It just failed at:

```
W0218 04:03:34.735] 2023/02/18 04:03:34 main.go:323: [Boskos] Fail To Release: 1 error occurred:
W0218 04:03:34.744]     * status 500 Internal Server Error, statusCode 500 releasing k8s-boskos-gce-project-03
W0218 04:03:34.744]
W0218 04:03:34.744] , kubetest err: <nil>
W0218 04:03:34.744] Traceback (most recent call last):
W0218 04:03:34.747]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 723, in <module>
W0218 04:03:34.747]     main(parse_args())
W0218 04:03:34.752]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 569, in main
W0218 04:03:34.752]     mode.start(runner_args)
W0218 04:03:34.752]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 228, in start
W0218 04:03:34.752]     check_env(env, self.command, *args)
W0218 04:03:34.752]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 111, in check_env
W0218 04:03:34.752]     subprocess.check_call(cmd, env=env)
W0218 04:03:34.752]   File "/usr/lib/python3.9/subprocess.py", line 373, in check_call
W0218 04:03:34.752]     raise CalledProcessError(retcode, cmd)
W0218 04:03:34.783] subprocess.CalledProcessError: Command '('kubetest', '--dump=/workspace/_artifacts', '--gcp-service-account=/etc/service-account/service-account.json', '--build=quick', '--stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gce-cos-alpha-features', '--up', '--down', '--test', '--provider=gce', '--cluster=e2e-9cdd61657a-92f64', '--gcp-network=e2e-9cdd61657a-92f64', '--ginkgo-parallel=1', '--extract=local', '--gcp-node-image=gci', '--gcp-zone=us-west1-b', '--runtime-config=api/all=true', '--test_args=--ginkgo.focus=\\[Feature:(GRPCContainerProbe|InPlacePodVerticalScaling|ProbeTerminationGracePeriod|APIServerTracing|APISelfSubjectReview|StorageVersionAPI|PodPreset|StatefulSetAutoDeletePVC|StatefulSetMinReadySeconds|ProxyTerminatingEndpoints|NodeOutOfServiceVolumeDetach|RetroactiveDefaultStorageClass|ReadWriteOncePod|RecoverVolumeExpansionFailure|CSINodeExpandSecret)\\]|Networking --ginkgo.skip=\\[Feature:(SCTPConnectivity|Volumes|Networking-Performance)\\]|IPv6|csi-hostpath-v0 --minStartupPods=8', '--timeout=180m')' returned non-zero exit status 1.
I0218 04:03:34.783] Done
E0218 04:03:34.784] Command failed
I0218 04:03:34.784] process 681 exited with code 1 after 74.1m
E0218 04:03:34.785] FAIL: pull-kubernetes-e2e-gce-cos-alpha-features
```

**Comment by Huang-Wei** [comment]
The error seems to originate from https://github.com/kubernetes/test-infra/blob/87e2d97fbf75c295508968f54e1f3853bdcadab8/kubetest/boskos/client/client.go#L497

**Comment by Huang-Wei** [comment]
@liggitt comments have been addressed. Could you please take another look? (e2e should be enabled by default now)

**Comment by Huang-Wei** [comment]
@ahg-g in case you want to take another look at scheduler changes (primarily tests)

**Comment by liggitt** [comment]
looks like the e2e was in the serial job, so spawned those... once those are green, lgtm

**Comment by Huang-Wei** [comment]
@liggitt the e2e passed in gce serial test suite:

image

**Pull request merged**